### PR TITLE
Manual Approval of Install Plans

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Implements install plan manual approval.

Addresses https://jira.coreos.com/browse/ALM-339

### CODE REVIEW

File: pkg/controller/operators/catalog/operator_test.go [hunk 1]
```diff
@@ -54,25 +54,35 @@ func TestTransitionInstallPlan(t *testing.T) {
...
 		transError error
+		approval   v1alpha1.Approval
+		approved   bool
 		expected   v1alpha1.InstallPlanPhase
 		condition  *v1alpha1.InstallPlanCondition
 	}{
-		{v1alpha1.InstallPlanPhaseNone, nil, v1alpha1.InstallPlanPhasePlanning, nil},
-		{v1alpha1.InstallPlanPhaseNone, err, v1alpha1.InstallPlanPhasePlanning, nil},
+		{v1alpha1.InstallPlanPhaseNone, nil, v1alpha1.ApprovalAutomatic, false, v1alpha1.InstallPlanPhasePlanning, nil},
+		{v1alpha1.InstallPlanPhaseNone, err, v1alpha1.ApprovalAutomatic, false, v1alpha1.InstallPlanPhasePlanning, nil},
```
Comment (ecordell) [inline pkg/controller/operators/catalog/operator_test.go]: just for completeness: `ApprovalAutomatic` + `approved = true` should go to planning as well

File: pkg/api/apis/installplan/v1alpha1/types.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ const (
 type InstallPlanSpec struct {
 	ClusterServiceVersionNames []string `json:"clusterServiceVersionNames"`
 	Approval                   Approval `json:"approval"`
+	Approved                   bool     `json:"approved"`
```
Comment (ecordell) [inline pkg/api/apis/installplan/v1alpha1/types.go]: This needs to be propagated to the CRD schema as well

Comment (alecmerdler) [reply pkg/api/apis/installplan/v1alpha1/types.go]: Which is the correct file with the `InstallPlan-v1` CRD? And how do I express that `approved` is only required if `approval: Manual`?

Comment (ecordell) [reply pkg/api/apis/installplan/v1alpha1/types.go]: The json-schema way is:

```json
  "dependencies": {
    "foo": ["bar"]
  }
```
(if foo is present, bar is required)

but IIRC the subset used for CRD validation doesn't allow that. It's worth checking, at least.

The canonical `InstallPlan` lives in `deploy/chart/templates`

File: deploy/chart/templates/06-installplan.crd.yaml [hunk 3]
```diff
@@ -41,3 +41,18 @@ spec:
               - Automatic
               - Manual
               - Update-Only # Will only apply an update if it updates existing packages only and doesn't add any new ones
+            approved:
+              type: boolean
+      anyOf:
```
Comment (alecmerdler) [inline deploy/chart/templates/06-installplan.crd.yaml]: Found this here: https://stackoverflow.com/questions/38717933/jsonschema-attribute-conditionally-required/38781027

Comment (ecordell) [reply deploy/chart/templates/06-installplan.crd.yaml]: We use that for the install strategy on CSV as well 👍

**Pull request merged**